### PR TITLE
plugins/nvim-lsp: Add a user option to set the cmd of each language server

### PR DESCRIPTION
**Pull request by GaetanLepage**
This is a proposal to implement the `plugins.lsp.servers.<name>.cmd` option.
This lets the user provide a custom command to run the language servers.
For some language servers, nixvim sets a default value for it. For others, it is left to `null`.

### CODE REVIEW

File: plugins/nvim-lsp/helpers.nix [hunk 1]
```diff
@@ -62,7 +66,7 @@
             {
               name = serverName;
               extraOptions = {
-                cmd = cmd cfg;
+                cmd = cfg.cmd;
```
Comment (pta2002) [inline plugins/nvim-lsp/helpers.nix:69]: Don't you have to pass in the argument here? so `cfg.cmg package`

Comment (pta2002) [reply plugins/nvim-lsp/helpers.nix:69]: nvm i got it

File: plugins/nvim-lsp/helpers.nix [hunk 2]
```diff
@@ -10,7 +10,7 @@
     serverName ? name,
     package ? pkgs.${name},
     extraPackages ? {},
-    cmd ? (cfg: null),
+    cmd ? (package: null),
```
Comment (pta2002) [inline plugins/nvim-lsp/helpers.nix:13]: Is there any reason you're changing it to `package` instead of `config`? The config seems more versatile

Comment (GaetanLepage) [reply plugins/nvim-lsp/helpers.nix:13]: I did that because I had an infinite recursion. But it might have been caused by something else...
Let me try again

Comment (pta2002) [reply plugins/nvim-lsp/helpers.nix:13]: Oh yeah i can see why it happens, since `cmd` is itself part of cfg. Maybe you can pass in `cfg` without `cmd`?

Comment (GaetanLepage) [reply plugins/nvim-lsp/helpers.nix]: Ok it seems to work !

**Review by pta2002 — APPROVED** [review]
This looks alright!
